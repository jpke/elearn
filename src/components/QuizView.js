import React from 'react';
import ListItems from './ListItems';

export const QuizView = (props) => (
  <div className="quizInProgress">
    <h3>Question {props.currentQuestionIndex + 1}</h3>
    <p>{props.question}</p>
    <h3>Answers</h3>
    <ul>
      <ListItems
        idSelected={props.idSelected}
        itemSelected={props.itemSelected}
        answers={props.answers}
        selectAnswer={props.selectAnswer}/>
    </ul>
    {props.currentQuestionIndex > 0 &&
        <button onClick={props.prevQuestion}>Previous Question</button>
    }
    {props.currentQuestionIndex != props.questionCount - 1 ? (
        <button onClick={props.nextQuestion}>Next Question</button>
      ) : (
        <div>
          <button onClick={props.submitQuiz}>Submit Quiz</button>
          <p>Unanswered Questions {props.unansweredQuestions}</p>
        </div>
      )
    }
  </div>
);
