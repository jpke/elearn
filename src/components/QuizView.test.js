/* eslint-disable no-undef */
/* eslint-disable import/no-named-as-default */

import React from 'react';
import renderer from 'react-test-renderer';

import QuizView from './QuizView';

let answers;
beforeAll(() => {
  answers =
  [
    {
      answer: "false current answer",
      correct: false
    },
    {
      answer: "true current answer",
      correct: true
    }
  ]
})

it('renders correctly quiz question without a selected answer', () => {
  const tree = renderer.create(
    <QuizView
      selectAnswer={jest.fn()}
      nextQuestion={jest.fn()}
      prevQuestion={jest.fn()}
      submitQuiz={jest.fn()}
      title={"sample quiz title"}
      question={"a quiz question here"}
      answers={answers}
      idSelected={""}
      itemSelected={""}
      currentQuestionIndex={0}
      questionCount={3}
      unansweredQuestions={3}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
})

it('renders correctly quiz question with a selected answer', () => {
  const tree = renderer.create(
    <QuizView
      selectAnswer={jest.fn()}
      nextQuestion={jest.fn()}
      prevQuestion={jest.fn()}
      submitQuiz={jest.fn()}
      title={"sample quiz title"}
      question={"a quiz question here"}
      answers={answers}
      idSelected={"0000"}
      itemSelected={0}
      currentQuestionIndex={0}
      questionCount={3}
      unansweredQuestions={3}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
})
