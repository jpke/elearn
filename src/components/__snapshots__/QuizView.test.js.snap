exports[`test renders correctly quiz question with a selected answer 1`] = `
<div
  className="quizInProgress">
  <h3
    className="title">
    sample quiz title
  </h3>
  <h3>
    Question 
    1
  </h3>
  <p>
    a quiz question here
  </p>
  <h3
    className="answers">
    Answers
  </h3>
  <ul
    className="item-list">
    <div>
      <li
        className="highlight"
        id={undefined}
        onClick={[Function]}>
        false current answer
      </li>
      <li
        className="none"
        id={undefined}
        onClick={[Function]}>
        true current answer
      </li>
    </div>
  </ul>
  <div
    className="quizNav-buttons">
    <button
      onClick={[Function]}>
      Next Question
    </button>
  </div>
</div>
`;

exports[`test renders correctly quiz question without a selected answer 1`] = `
<div
  className="quizInProgress">
  <h3
    className="title">
    sample quiz title
  </h3>
  <h3>
    Question 
    1
  </h3>
  <p>
    a quiz question here
  </p>
  <h3
    className="answers">
    Answers
  </h3>
  <ul
    className="item-list">
    <div>
      <li
        className="none"
        id={undefined}
        onClick={[Function]}>
        false current answer
      </li>
      <li
        className="none"
        id={undefined}
        onClick={[Function]}>
        true current answer
      </li>
    </div>
  </ul>
  <div
    className="quizNav-buttons">
    <button
      onClick={[Function]}>
      Next Question
    </button>
  </div>
</div>
`;
